src/efftester.ml

```diff
@@ -1598,66 +1595,69 @@ let unify_funtest =
         types_compat sty sty' || types_compat sty' sty)
 ;;
 
-(* FIXME: update to dep_term_gen *)
+(** Helpers *)
+
+let make_logger file_path =
+  let file_out = open_out file_path in
+  let counter = ref 0 in
+  fun fmt ->
+    incr counter;
+    Printf.fprintf file_out ("(* %d *) " ^^ fmt ^^ ";;\n%!") !counter
+;;
+
+let no_logger =
+  let discard_out = open_out "/dev/null" in
+  Printf.fprintf discard_out
```

Comment:
You can use `Printf.ifprintf stdout`. [ifprintf](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Printf.html#VALifprintf) requires a channel argument and a format, it consumes the format argument but then does nothing, so you don't need to open `/dev/null` (which may not always exists, eg. on Windows).